### PR TITLE
[FB] PMM-6119 dbaas-controller packaging

### PR DESCRIPTION
**Pull request by Sevlin**
Test pipeline on [pmm.cd](https://pmm.cd.percona.com/job/pmm2-server-autobuild-test-dbaas/)

Related PR's:
- [x] https://github.com/percona/pmm-server-packaging/pull/146
- [x] https://github.com/Percona-Lab/jenkins-pipelines/pull/491
- [x] https://github.com/Percona-Lab/percona-images/pull/77
- [x] https://github.com/Percona-Lab/percona-images/pull/80
- [x] https://github.com/percona/pmm-update/pull/94

### CODE REVIEW

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-956-94912b3
client docker - perconalab/pmm-client-fb:PR-956-94912b3
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-956-94912b3.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-956-934246a
client docker - perconalab/pmm-client-fb:PR-956-934246a
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-956-934246a.tar.gz

**Comment by AlekSi** [comment]
`release/0.0.2` is in the base branch now

**Comment by Sevlin** [comment]
Latest build: https://pmm.cd.percona.com/job/pmm2-autobuild-test/7/

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-956-a0c9f5b
client docker - perconalab/pmm-client-fb:PR-956-a0c9f5b
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-956-a0c9f5b.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-956-aab5f32
client docker - perconalab/pmm-client-fb:PR-956-aab5f32
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-956-aab5f32.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-956-e3d048c
client docker - perconalab/pmm-client-fb:PR-956-e3d048c
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-956-e3d048c.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-956-ee37dec
client docker - perconalab/pmm-client-fb:PR-956-ee37dec
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-956-ee37dec.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-956-1dfc54e
client docker - perconalab/pmm-client-fb:PR-956-1dfc54e
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-956-1dfc54e.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-956-5029580
client docker - perconalab/pmm-client-fb:PR-956-5029580
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-956-5029580.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-956-eaee617
client docker - perconalab/pmm-client-fb:PR-956-eaee617
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-956-eaee617.tar.gz

**Comment by Sevlin** [comment]
> pmm-update changes were not added to this FB

Updated **pmm-update** submodule.

**Comment by AlekSi** [comment]
> + make PMM_RELEASE_VERSION=2.9.0-HEAD-0c4e925 > PMM_RELEASE_FULLCOMMIT=bf016ec9dfae115f0a513653b68c82ca1e4d1ed1 PMM_RELEASE_FULLCOMMIT= release
fatal: Not a git repository (or any of the parent directories): .git

We should copy `.git` too.

```
bin/dbaas-controller --version
panic: dbaas-controller version is not set during build.

goroutine 1 [running]:
main.main()
	/home/builder/rpm/BUILD/dbaas-controller-bf016ec9dfae115f0a513653b68c82ca1e4d1ed1/cmd/dbaas-controller/main.go:36 +0x5fa
make: *** [release] Error 2
error: Bad exit status from /var/tmp/rpm-tmp.FTR4G6 (%build)
    Bad exit status from /var/tmp/rpm-tmp.FTR4G6 (%build)


RPM build errors:
++ dirname /mnt/workspace/pmm-submodules_PR-956/build/bin/build-server-rpm
```

That should fail that build step, but it passes: https://pmm.cd.percona.com/blue/organizations/jenkins/pmm-submodules/detail/PR-956/20/pipeline/71

**Comment by AlekSi** [comment]
And why `PMM_RELEASE_FULLCOMMIT` is added twice?

**Comment by Sevlin** [comment]
I've forgot to remove extra `PMM_RELEASE_FULLCOMMIT` when I was troubleshooting build process

**Comment by EvgeniyPatlan** [comment]
> We should copy `.git` too.

No. We should teach Makefile to work without .git. We never put .git repository into source tarballs.

**Comment by AlekSi** [comment]
> We never put .git repository into source tarballs.

So I just downloaded https://www.percona.com/downloads/pmm2/2.9.0/source/debian/pmm2-client_2.9.0.orig.tar.gz from https://www.percona.com/downloads/pmm2/ – it contains `.git` directory.

But I also said nothing about having `.git` directory in the source tarball. But we do need it for dbaas-controller build – there is no other way to set dbaas-controller version (as opposed to PMM version) because it has its own version in addition to PMM version because it is not always a part of PMM.

**Comment by AlekSi** [comment]
And please also make a change to break the build at this stage, not just skip the error.

**Comment by EvgeniyPatlan** [comment]
> > We never put .git repository into source tarballs.
> 
> So I just downloaded https://www.percona.com/downloads/pmm2/2.9.0/source/debian/pmm2-client_2.9.0.orig.tar.gz from https://www.percona.com/downloads/pmm2/ – it contains `.git` directory.
> 
> But I also said nothing about having `.git` directory in the source tarball. But we do need it for dbaas-controller build – there is no other way to set dbaas-controller version (as opposed to PMM version) because it has its own version in addition to PMM version because it is not always a part of PMM.

while downloading sources we can get needed version and store it in some file. While building export variable from this file

**Comment by AlekSi** [comment]
> while downloading sources we can get needed version and store it in some file. While building export variable from this file

Great, let's do it. All variables are there: https://github.com/percona-platform/dbaas-controller/blob/87e11c69068c6b3c3303de22e597082a3aa77f73/Makefile#L10-L17

But other comments were not addressed.

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-956-371e528
client docker - perconalab/pmm-client-fb:PR-956-371e528
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-956-371e528.tar.gz

File: build/bin/build-server-docker [hunk 1]
```diff
@@ -12,6 +12,9 @@ fi
 
 cp -r ${root_dir}/sources/percona-images/ansible ${rpms_dir}/../ansible
 
+# For testing purposes only
```
Comment (AlekSi) [inline build/bin/build-server-docker:15]: So I guess it should be reverted?

File: build/pmm-server-docker/Dockerfile [hunk 2]
```diff
@@ -5,15 +5,16 @@ EXPOSE 80 443
 WORKDIR /opt
 
 RUN useradd -s /bin/false pmm
-RUN yum -y install epel-release && yum -y install ansible
+RUN yum -y install epel-release \
+    && yum -y install ansible
```
Comment (AlekSi) [inline build/pmm-server-docker/Dockerfile:9]: There is not real need to use `&&` – we squash all image layers

File: build/pmm-server-docker/Dockerfile [hunk 3]
```diff
@@ -5,15 +5,16 @@ EXPOSE 80 443
...
 
 COPY RPMS /tmp/RPMS
 COPY gitCommit /tmp/gitCommit
 
 COPY ansible /opt/ansible
 RUN cp -r /opt/ansible/roles /opt/ansible/pmm2-docker/roles
-RUN ansible-playbook -vvv -i 'localhost,' -c local /opt/ansible/pmm2-docker/main.yml \
-    && ansible-playbook -vvv -i 'localhost,' -c local /usr/share/pmm-update/ansible/playbook/tasks/update.yml \
+RUN ansible-playbook -vvv -i 'localhost,' -c local -e 'pmm2_server_repo=laboratory' /opt/ansible/pmm2-docker/main.yml \
+    && ansible-playbook -vvv -i 'localhost,' -c local -e 'pmm2_server_repo=laboratory' /usr/share/pmm-update/ansible/playbook/tasks/update.yml \
```
Comment (AlekSi) [inline build/pmm-server-docker/Dockerfile:17]: How that will affect main branch autobuilds?

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-956-6c9512d
client docker - perconalab/pmm-client-fb:PR-956-6c9512d
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-956-6c9512d.tar.gz

File: build/pmm-server-docker/Dockerfile (hunk 2 above)
Comment (Sevlin) [reply build/pmm-server-docker/Dockerfile:9]: This change will be reverted. It was trying to fix installation issue by adding priorities plugin for yum.

File: build/pmm-server-docker/Dockerfile (hunk 3 above)
Comment (Sevlin) [reply build/pmm-server-docker/Dockerfile:17]: As I said, these changes are temporary and they won't be used in upstream. Just an override for playbook vairable in scope of this FB.

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-956-53f5f61
client docker - perconalab/pmm-client-fb:PR-956-53f5f61
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-956-53f5f61.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-956-cf9154d
client docker - perconalab/pmm-client-fb:PR-956-cf9154d
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-956-cf9154d.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-956-b4cb8fe
client docker - perconalab/pmm-client-fb:PR-956-b4cb8fe
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-956-b4cb8fe.tar.gz

**Pull request closed without merging**